chat-widget/src/components/livechatwidget/common/setPostChatContextAndLoadSurvey.ts

```diff
@@ -33,4 +42,205 @@ export const setPostChatContextAndLoadSurvey = async (chatSDK: any, dispatch: Di
     BroadcastService.getMessageByEventName("LoadPostChatSurvey").subscribe((msg: ICustomEvent) => {
         dispatch({ type: LiveChatWidgetActionType.SET_CONVERSATION_STATE, payload: ConversationState.Postchat });
     });
-};
\ No newline at end of file
+};
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const checkPostChatEnabled = (props: ILiveChatWidgetProps, state: ILiveChatWidgetContext) => {
+    const isPostChatEnabled = props.chatConfig?.LiveWSAndLiveChatEngJoin?.msdyn_postconversationsurveyenable ?? state.domainStates.liveChatConfig?.LiveWSAndLiveChatEngJoin?.msdyn_postconversationsurveyenable;
+    if (isPostChatEnabled !== Constants.true) {
+        return false;
+    }
+    return true;
+};
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const initiatePostChat = async (props: ILiveChatWidgetProps, chatSDK: any, setAdapter: any, setWebChatStyles: any, dispatch: Dispatch<ILiveChatWidgetAction>, adapter: any, state: ILiveChatWidgetContext) => {
+    const postChatSurveyMode = props.chatConfig?.LiveWSAndLiveChatEngJoin?.msdyn_postconversationsurveymode ?? state.domainStates.liveChatConfig?.LiveWSAndLiveChatEngJoin?.msdyn_postconversationsurveymode;
```

Comment:
Why are we only checking the mode for agent survey not bot survey here? Shouldn't we decide after getting ConversationDetails?